src/LinuxTracing/LeafFunctionCallManagerTest.cpp

```diff
@@ -0,0 +1,339 @@
+              (pid_t, unwindstack::Maps*, (const std::array<uint64_t, PERF_REG_X86_64_MAX>&),
+               const void*, uint64_t, bool, size_t),
+              (override));
+};
+
+class LeafFunctionCallManagerTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kUprobesMapsStart), Lt(kUprobesMapsEnd))))
+        .WillRepeatedly(Return(&kUprobesMapInfo));
+
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kTargetMapsStart), Lt(kTargetMapsEnd))))
+        .WillRepeatedly(Return(&kTargetMapInfo));
+
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kNonExecutableMapsStart), Lt(kNonExecutableMapsEnd))))
+        .WillRepeatedly(Return(&kNonExecutableMapInfo));
+  }
+
+  void TearDown() override {}
+
+  MockLibunwindstackMaps maps_;
+  MockLibunwindstackUnwinder unwinder_;
+
+  LeafFunctionCallManager leaf_function_call_manager_;
+
+  static constexpr uint64_t kUprobesMapsStart = 42;
+  static constexpr uint64_t kUprobesMapsEnd = 84;
+
+  static constexpr uint64_t kTargetMapsStart = 100;
+  static constexpr uint64_t kTargetMapsEnd = 200;
+
+  static constexpr uint64_t kNonExecutableMapsStart = 500;
+  static constexpr uint64_t kNonExecutableMapsEnd = 600;
+
+  static constexpr uint64_t kKernelAddress = 11;
+
+  static constexpr uint64_t kTargetAddress1 = 100;
+  static constexpr uint64_t kTargetAddress2 = 200;
+  static constexpr uint64_t kTargetAddress3 = 300;
+
+  static inline const std::string kUprobesName = "[uprobes]";
+  static inline const std::string kTargetName = "target";
+  static inline const std::string kNonExecutableName = "data";
+
+  static inline unwindstack::MapInfo kUprobesMapInfo{
+      nullptr, kUprobesMapsStart, kUprobesMapsEnd, 0, PROT_EXEC | PROT_READ, kUprobesName};
+
+  static inline unwindstack::MapInfo kTargetMapInfo{nullptr, kTargetMapsStart,      kTargetMapsEnd,
+                                                    0,       PROT_EXEC | PROT_READ, kTargetName};
+
+  static inline unwindstack::MapInfo kNonExecutableMapInfo{
+      nullptr, kNonExecutableMapsStart, kNonExecutableMapsEnd, 0, PROT_READ, kNonExecutableName};
+
+  static inline unwindstack::FrameData kFrame1{
+      .pc = kTargetAddress1,
+      .function_name = "foo",
+      .function_offset = 0,
+      .map_name = kTargetName,
+  };
+
+  static inline unwindstack::FrameData kFrame2{
+      .pc = kTargetAddress2,
+      .function_name = "bar",
+      .function_offset = 0,
+      .map_name = kTargetName,
+  };
+
+  static inline unwindstack::FrameData kFrame3{
+      .pc = kTargetAddress3,
+      .function_name = "baz",
+      .function_offset = 0,
+      .map_name = kTargetName,
+  };
+
+  static inline unwindstack::FrameData kNonExecutableFrame{
+      .pc = kNonExecutableMapsStart,
+      .function_name = "???",
+      .function_offset = 0,
+      .map_name = kNonExecutableName,
+  };
+};
+
+}  // namespace
+
+TEST_F(LeafFunctionCallManagerTest, PatchLeafFunctionCallerReturnsErrorOnSmallStackSamples) {
```

Comment:
Yeah, now all these tests should be renamed changing "PatchLeafFunctionCaller" to "PatchCallerOfLeafFunction".

Reply:
Done.